<launch>
 <env name="GAZEBO_MODEL_PATH" value="$(find irobotcreate)/models/"/>
 <param name="robot_description" textfile="$(find irobotcreate)/models/irobot/model_full.urdf"/>

 <!-- load an empty gazebo world with standard physics -->
 <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find RoboMap)/worlds/new_mapping.world"/>
 </include>

 <node name ="tf_broadcast" pkg="irobotcreate" type="tf_broadcast"/>

     <!-- robot one -->
    <!--<group ns="robot1">-->
     <node name="spawn_model"
        pkg="gazebo_ros"
        type="spawn_model"
        args="-file $(find irobotcreate)/models/irobot/model_full.sdf -sdf -model iRobot1 -x 2.5 -y 3 -z 0"/>
     <!--<node name="detect" pkg="RoboMap" type="detect"/>
     <node name="seek_node" pkg="RoboMap" type="seek_node"/>-->
     <node name="arbiter_node" pkg="RoboMap" type="arbiter_node"/>
     <!--<node name="wander_node" pkg="RoboMap" type="wander_node"/>-->
     <node name="avoid_node" pkg="RoboMap" type="avoid_node"/>
    <!--</group>-->

    
    <!--<group ns="robot2">
     <node name="spawn_model"
        pkg="gazebo_ros"
        type="spawn_model"
        args="-file $(find irobotcreate)/models/irobot/model_full.sdf -sdf -model iRobot2 -x 2 -y 16 -z 0"/>
     <node name="detect" pkg="RoboMap" type="detect"/>
     <node name="seek_node" pkg="RoboMap" type="seek_node"/>
     <node name="arbiter_node" pkg="RoboMap" type="arbiter_node"/>
     <node name="wander_node" pkg="RoboMap" type="wander_node"/>
     <node name="avoid_node" pkg="RoboMap" type="avoid_node"/>
    </group>

    

    <group ns="robot3">
     <node name="spawn_model"
        pkg="gazebo_ros"
        type="spawn_model"
        args="-file $(find irobotcreate)/models/irobot/model_full.sdf -sdf -model iRobot3 -x 15 -y 15 -z 0"/>
     <node name="detect" pkg="RoboMap" type="detect"/>
     <node name="seek_node" pkg="RoboMap" type="seek_node"/>
     <node name="arbiter_node" pkg="RoboMap" type="arbiter_node"/>
     <node name="wander_node" pkg="RoboMap" type="wander_node"/>
     <node name="avoid_node" pkg="RoboMap" type="avoid_node"/>
    </group>-->

 <node pkg="irobotcreate" type="gps_to_zero_translate" name="gps_to_zero_translate"/>

 <!--<node name="mapNode" pkg="RoboMap" type="mapNode"/>-->


 <!-- EKF for absolute localization (clearpath) -->
 <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf" output="screen">
     <param name="base_footprint_frame" value="base_link"/>
     <remap from="odom" to="irobot/odom"/>
     <remap from="imu_data" to="irobot/imu_data"/>
     <remap from="gps" to="irobot/gps_odom"/>
     <param name="output_frame" value="odom"/>
     <param name="freq" value="10.0"/>
     <param name="sensor_timeout" value="1.0"/>
     <param name="odom_used" value="true"/>
     <param name="imu_used" value="true"/>
     <param name="imu_absolute" value="true"/>
     <param name="vo_used" value="false"/>
     <param name="gps_used" value="true"/>
     <param name="debug" value="false"/>
     <param name="self_diagnose" value="false"/>

 </node>
</launch>